src/models/connectionStore.ts

```diff
@@ -103,10 +126,26 @@ export class ConnectionStore {
         );
     }
 
+    /**
+     * Creates a credential ID using profile ID instead of connection details
+     * @param profileId The connection profile ID
+     * @param itemType The item type string
+     * @returns formatted credential ID
+     */
+    public static formatCredentialIdFromProfileId(
+        profileId: string,
+        itemType: string = ConnectionStore.CRED_PROFILE_USER,
+    ): string {
+        let cred: string[] = [ConnectionStore.CRED_PREFIX];
+        ConnectionStore.pushIfNonEmpty(profileId, ConnectionStore.CRED_PROFILE_PREFIX, cred);
+        ConnectionStore.pushIfNonEmpty(itemType, ConnectionStore.CRED_ITEMTYPE_PREFIX, cred);
+        return cred.join(ConnectionStore.CRED_SEPARATOR);
+    }
+
     /**
      * Creates a formatted credential usable for uniquely identifying a SQL Connection.
      * This string can be decoded but is not optimized for this.
-     * @static
+     * @deprecated
```

Comment:
add deprecation note indicating that callers should use `formatCredentialIdFromProfileId` instead